src/ui/public/agg_response/tabify/_response_writer.js

```diff
@@ -25,22 +25,28 @@ import { tabifyGetColumns } from './_get_columns';
  * produces a table, or a series of tables.
  *
  * @param {AggConfigs} aggs - the agg configs object to which the aggregation response correlates
- * @param {boolean} metricsAtAllLevels - setting to true will produce metrics for every bucket
- * @param {boolean} partialRows - setting to true will not remove rows with missing values
+ * @param {boolean} minimalColumns - setting to true will only return a column for the last bucket/metric instead of one for each level
+ * @param {boolean} partialRows - vis.params.showPartialRows: determines whether to return rows with incomplete data
+ * @param {Object} timeRange - time range object, if provided
  */
-function TabbedAggResponseWriter(aggs, { metricsAtAllLevels = false, partialRows = false, timeRange } = {}) {
+function TabbedAggResponseWriter(aggs, {
+  minimalColumns = true,
+  partialRows = false,
+  timeRange
+} = {}) {
+  // Private
+  this._removePartialRows = !partialRows;
+  this._minimalColumns = minimalColumns;
```

Comment:
Is there a specific reason we're storing that on a private variable? It seems to be only used locally in this function?

Reply:
The goal here was to prevent confusion - I was surprised to find that some of the properties of the response writer were public (used in `tabify.js`) 

So the intent was to make it clear which items were part of the response writer’s contract, and which were just local implementation details and could be safely altered.